JS.md

```diff
@@ -0,0 +1,35 @@
+# JS client-side conventions
+
+JavaScript is often difficult to get started with because the landscape (and
+language) is vast, and the community has less adherence to conventions.
+
+This is a conventions document for not getting into trouble in JS.
+
+In the future, we will have a little js pack repo for copying in a good set
+of standards into a new application.
+
+## React
+
+* Use `jsx` to build templates
+* Create templates that are as declarative as possible
+* Extract all logic into presenters
+
+## Redux
+
+* Use `react-redux` to bind state and handlers to jsx templates
+* Put all state into the redux store
+* Even UI related state. Once this is a pattern it becomes easy
+* Don't use `jsx` in the connect files, instead create a new template
+* Extract logic for handlers into a handlers area
+* Extract logic reasoning about state into presenters
+
+## Testing
+
+* [jest](https://jestjs.io/) - for unit testing
+* [cypress](https://www.cypress.io/) - for acceptance testing
```

Comment:
Should we be buying the paid Cypress thingie, or is the free version sufficient for our purposes?